src/main/scala/com/example/solution/services/Screenings.scala

```diff
@@ -0,0 +1,153 @@
+package com.example.solution.services
+
+import cats.effect._
+import cats.syntax.all._
+import com.example.solution.domain.room.{Room, RoomId}
+import com.example.solution.domain.screening.Screening
+import com.example.solution.dto.reservation.ScreeningInterval
+import com.example.solution.dto.screening.{AvailableSeat, RoomData, ScreeningData}
+import skunk._
+import skunk.codec.all._
+import skunk.implicits._
+
+import java.time.LocalDateTime
+
+trait Screenings[F[_]] {
+
+  def listScreenings(interval: ScreeningInterval): F[List[ScreeningData]]
+
+  def pickScreening(screeningId: Int): F[Option[RoomData]]
+
+}
+
+object LiveScreenings {
+  def make[F[_] : Sync](
+                         sessionPool: Resource[F, Session[F]]
+                       ): F[Screenings[F]] =
+    Sync[F].delay(
+      new LiveScreenings[F](sessionPool)
+    )
+}
+
+final class LiveScreenings[F[_] : BracketThrow : Sync] private(
+                                                                sessionPool: Resource[F, Session[F]]
+                                                              ) extends Screenings[F] {
+
+  import ScreeningQueries._
+
+  override def listScreenings(interval: ScreeningInterval): F[List[ScreeningData]] =
+    sessionPool.use { session => {
+      session.prepare(selectAllInRange).use { ps =>
+        ps.stream((interval.start, interval.finish), chunkSize = 1024).compile.toList
+      }.map(listScreenings => listScreenings.sorted)
+    }
+    }
+
+
+  override def pickScreening(screeningId: Int): F[Option[RoomData]] = sessionPool.use { session =>
+    session.prepare(selectRoomInfo).use { ps =>
+      ps.stream(screeningId, 1024).compile.toList
+        .map(convertToRoomData)
+    }
+  }
+
+
+  private def convertToRoomData(list: List[Either[SimpleRoomData, FullRoomData]]): Option[RoomData] = {
+    if (list.isEmpty) None
+    else if (list.exists(_.isRight))
+      Some(handleFullRoomData(list))
+    else
+      list.collectFirst {
+        case Left(value) => RoomData(
+          Room(RoomId(value.roomId), value.roomRows, value.roomEatsPerRow),
+          for {
+            row <- (1 to value.roomRows).toList
+            seatInRow <- (1 to value.roomEatsPerRow).toList
+          } yield AvailableSeat(row, seatInRow),
+          value.screeningTime
+        )
+      }
+
+
+  }
+
+  private def handleFullRoomData(list: List[Either[SimpleRoomData, FullRoomData]]): RoomData = {
+    val takenSeats = list.collect {
+      case Right(fullRoomData) => (fullRoomData.takenRow, fullRoomData.takenSeatRow)
+    }
+    list.collectFirst {
+      case Right(value) => RoomData(
+        Room(RoomId(value.roomId), value.roomRows, value.roomSeatsPerRow),
+        (for {
+          row <- (1 to value.roomRows).toList
+          seatInRow <- (1 to value.roomSeatsPerRow).toList
+        } yield AvailableSeat(row, seatInRow))
+          .filter(as => !takenSeats.contains(as.row, as.seatInRow)),
+        value.screeningTime
+      )
+    }.get
```

Comment:
how would you avoid calling `get`?

Reply:
Since we know that at least one Right value exists(list.exists(_.isRight) in convertToRoomData), we could pass it explicitly as another argument to the function